gamemode/modules/rafts/sh_raft_class.lua

```diff
@@ -0,0 +1,166 @@
+local raftMeta = {}
+raftMeta.__index = raftMeta
+RVR.raftsList = {}
+
+function raftMeta:AddPiece( position, ent )
+    ent:SetRaftGridPosition( position )
+    self.pieces[ent:EntIndex()] = ent
+    self.grid[self.vectorIndex( position )] = ent:EntIndex()
+
+    net.Start("RVR_Raft_NewRaftPiece")
+        net.WriteInt( self.id, 32 )
+        net.WriteInt( ent:EntIndex(), 32 )
+        net.WriteVector( position )
+    net.Broadcast()
+
+    -- TODO only owners of the raft should get this info
+end
+
+function raftMeta:RemovePiece( ent )
+    self.pieces[ent:EntIndex()] = nil
+end
+
+function raftMeta:GetPiece( position )
+    local index = self.grid[self.vectorIndex( position )]
+    if not index then return end
+
+    local ent = self.pieces[index]
+
+    if not IsValid( ent ) then
+        self.pieces[index] = nil
+        return
+    end
+    return ent
+end
+
+function raftMeta:GetNeighbors( piece )
+    local neighbors = {}
+
+    for x=-1, 1 do
+        for y=-1, 1 do
+            for z=-1, 1 do
+                local dir = Vector( x, y, z )
+                if dir:IsZero() then
```

Comment:
```suggestion
                if not dir:IsZero() then
```

Reply:
i fixed this earlier but forgot to push the changes

Reply:
O oke